### PR TITLE
fix-new-chat

### PR DESCRIPTION
**Pull request by johndev0711**
<If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit.>

### Details
<Explanation of the change or anything fishy that is going on>
There are 4 requirements about this GitHub issue, that is, 
1. "Remove the feedback... ", 
2. "When typing... ... ... show what they've typed below, but greyed out and unselectable.", 
3. "... ... ... show the option as selectable(as we do today)", 
4. "Phone number validation... 'please enter... ... ... the country code...'"

All of them are related with the **getHeaderMessage** function in **OptionsListUtils.js**.

This **getHeaderMessage** function has 2 props as the original props: **hasSelectableOptions**, **hasUserToInvite**.
We have to add a props - **searchValue** to them, this used to implement the second and fourth requirements.
Also, we have to change the **createNewChat** function of the **NewChatPage.js** by adding **this.state.searchValue**. Here **this.state.searchValue** has the inputted value. And this value show the feedback according to the second requirement.

For solving the fourth requirement, we have to do validation with the inputted value (**searchValue**).
This problem has been solved easily by using simple JavaScript validation code.

Additionally, I've changed the default messages in **CONST.js** with the message of the fourth requirement.
### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>

https://github.com/Expensify/Expensify.cash/issues/1968

### Tests

Running the web app: npm run build/npm run web
Running the MacOS desktop app: npm run desktop-build/npm run desktop
Running the iOS app: npm run ios-build/npm run ios
Android app: npm run android-build /npm run android
1. Run and click the **(+) /New Chat** button.
2. Click on the text input to bring it into focus
3. Type the text which exist on the list, then the below list would be filtered. (Ex: "toti")
4. When typing, if the input does not match a valid email or phone number format then show what they've typed below, but greyed out and unselectable. Once the input matches a valid email or phone number format then show the option as selectable (as we do today)
  For example, **abcde@fgs** : _show what they've typed below, but greyed out and unseletable._
**abcde@fgs.com** : _show the option as selectable._
6. If the input is only numbers and begins with 0, then display message **Please enter a phone number including the country code e.g +447814266907**
For example, **123..(.number)** => _Show the selectable phone number,_ **012...(number)** => _Show message **please enter ...**,_ 

All testing results have added as following attachments(screenshots)

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

image
image
image
image
image
image
image
image
image

#### Mobile Web

image
image
image
image
image
image
image
image

#### Desktop

image
image
image
image
image
image

#### iOS

image

image
image

image

image

image

image

#### Android

image
image
image
image
image
image
image

### CODE REVIEW

**Comment by johndev0711** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by puneetlath** [comment]
Hey @johndev0711 can you please fill in the PR template? You need to link the GH issue that your PR is for as well as provide instructions on how to test your PR and screenshots from each of the platforms listed to show that it works on each platform. Thanks!

**Comment by johndev0711** [comment]
> Hey @johndev0711 can you please fill in the PR template? You need to link the GH issue that your PR is for as well as provide instructions on how to test your PR and screenshots from each of the platforms listed to show that it works on each platform. Thanks!

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -407,16 +407,21 @@ function getSidebarOptions(reports, personalDetails, draftComments, activeReport
...
  */
-function getHeaderMessage(hasSelectableOptions, hasUserToInvite, maxParticipantsReached = false) {
+function getHeaderMessage(hasSelectableOptions, hasUserToInvite, searchValue, maxParticipantsReached = false) {
     if (maxParticipantsReached) {
         return CONST.MESSAGES.MAXIMUM_PARTICIPANTS_REACHED;
     }
 
     if (!hasSelectableOptions && !hasUserToInvite) {
-        return CONST.MESSAGES.NO_CONTACTS_FOUND;
+        if (/^\d+$/.test(searchValue)) {
```
Comment (Dal-Papa) [inline src/libs/OptionsListUtils.js:420]: Does this check if it starts with a `0` ? I don't see it here.

Comment (cocoa-1012) [reply src/libs/OptionsListUtils.js:420]: If it starts with a 0, the country code validation function runs, the function is already on source code.
Thanks for your kindness.

Comment (johndev0711) [reply src/libs/OptionsListUtils.js:420]: Yes, you're right.
If it starts with a '0', the country code validation function runs. The function exist already in source code.
Thanks for your kindness.

**Review by Dal-Papa — APPROVED** [review]

File: src/libs/OptionsListUtils.js [hunk 2]
```diff
@@ -407,16 +407,21 @@ function getSidebarOptions(reports, personalDetails, draftComments, activeReport
  *
  * @param {Boolean} hasSelectableOptions
  * @param {Boolean} hasUserToInvite
+ * @param {String} searchValue
  * @param {Boolean} [maxParticipantsReached]
  * @return {String}
  */
-function getHeaderMessage(hasSelectableOptions, hasUserToInvite, maxParticipantsReached = false) {
+function getHeaderMessage(hasSelectableOptions, hasUserToInvite, searchValue, maxParticipantsReached = false) {
```
Comment (marcaaron) [inline src/libs/OptionsListUtils.js:414]: If we change a method signature we must update all usages. 🙃

Comment (johndev0711) [reply src/libs/OptionsListUtils.js:414]: Oh, yes, I've just checked it.
I've thought the only newChat function.
Your concern can be solved easily by adding a props(this.state.searchValue) in the NewGroupPage.js.
image
Thanks.

**Pull request merged**